tests/test_tokenization_common.py

```diff
@@ -2110,6 +2110,71 @@ def test_batch_encode_plus_batch_sequence_length(self):
                         encoded_sequences_batch_padded_2[key],
                     )
 
+    def test_added_token_are_never_split(self):
+        if not self.test_slow_tokenizer:
+            self.skipTest("Currently this test is only for slow tokenizers")
+            return
+        model_ids = self.tokenizer_class.pretrained_vocab_files_map[self.from_pretrained_vocab_key].keys()
+        tokenizer = self.tokenizer_class.from_pretrained(list(model_ids)[0])
+        new_tokens = []
+        new_tokens.append(AddedToken("<lstrip=False, rstrip=False>", lstrip=False, rstrip=False))
+        new_tokens.append(AddedToken("<lstrip=True, rstrip=False>", lstrip=True, rstrip=False))
+        new_tokens.append(AddedToken("<lstrip=False, rstrip=True>", lstrip=False, rstrip=True))
+        new_tokens.append(AddedToken("<lstrip=True, rstrip=True>", lstrip=True, rstrip=True))
+
+        for token in new_tokens:
+            with self.subTest(f"testing with {token.content[1:-1]}"):
+                space = tokenizer.tokenize(" ")[0]
+                if len(space) > 1:
+                    # BPE adds a spiece underline
+                    space = space[-1]
```

Comment:
This is not really fullproof, as `Bert` for example always eats away ending spaces like `My name is ` is tokenized as `... ,'#is']` . Thus even if `lstrip` is set to False, the space will be stripped.